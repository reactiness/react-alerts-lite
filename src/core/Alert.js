import React, { Component } from "react";
// import { actions } from './actions';
import {
  defaultStyle,
  containerStyle,
  alertStyles,
  iconStyle,
  closeButtonStyle,
  contentStyle
} from "../css/styles";
import info from "../assets/info.svg";
import close from "../assets/x.svg";

export class Alert extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      renderChild: true
    };
    this.interval = null;
  }

  componentDidMount() {
    this.interval = setTimeout(
      () => this.props.remove(this.props.id),
      this.props.timeout
    );
  }

  componentWillUnmount() {
    const { onClose } = this.props;
    if (onClose) onClose();
    clearInterval(this.interval);
  }

  render() {
    const {
      content,
      type,
      remove,
      style: transitionStyle,
      icon,
      ...props
    } = this.props;
    return (
      <div style={{ ...containerStyle, ...transitionStyle }} {...props}>
        <div
          style={{
            ...defaultStyle,
            ...alertStyles[type]
          }}
        >
          <div style={iconStyle}>
            <img src={info} style={{ margin: "0 auto" }} />
          </div>
          <div style={contentStyle}>{content}</div>
          <div style={closeButtonStyle}>
            <img src={close} style={{ margin: "0 auto" }} />
          </div>
        </div>
      </div>
    );
  }
}
