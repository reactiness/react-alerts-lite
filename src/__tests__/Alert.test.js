import React from "react";
import { Alert } from "../core/Alert.js";

describe("Alert", () => {
  it("should be defined", () => {
    expect(Alert).toBeDefined();
  });

  it("should render correctly", () => {
    const wrapper = shallow(
      <Alert
        content="this is a test alert"
        type="success"
        remove={jest.fn()}
        closeButton={true}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div.__rta_container")).toHaveLength(1);
    expect(wrapper.find("div.__rta_icon")).toHaveLength(1);
    expect(wrapper.find("div.__rta_content")).toHaveLength(1);
    expect(wrapper.find("div.__rta_close")).toHaveLength(1);
  });
  it("should render without close icon when closeButton prop equals false", () => {
    const wrapper = shallow(
      <Alert
        content="this is a test alert"
        type="success"
        remove={jest.fn()}
        closeButton={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div.__rta_close")).toHaveLength(0);
  });
  it("should render success alert correctly", () => {
    const wrapper = shallow(
      <Alert content="test" type="success" remove={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render warning alert correctly", () => {
    const wrapper = shallow(
      <Alert content="test" type="warning" remove={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render error alert correctly", () => {
    const wrapper = shallow(
      <Alert content="test" type="error" remove={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render basic alert correctly", () => {
    const wrapper = shallow(
      <Alert content="test" type="basic" remove={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should render info alert correctly", () => {
    const wrapper = shallow(
      <Alert content="test" type="info" remove={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should call remove on clicking close button", () => {
    const remove = jest.fn();
    const wrapper = mount(<Alert content="test" type="info" remove={remove} closeButton={true} />);
    wrapper.find("div.__rta_close").simulate("click");
    expect(remove).toHaveBeenCalled();
  });
});

// closeButton: true,
// content: "this is a test alert",
// type: "basic",
