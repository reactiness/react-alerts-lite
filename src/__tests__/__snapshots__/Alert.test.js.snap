// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render basic alert correctly 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(192,192,192,0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      test
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly from snapshot 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(76, 175, 80, 0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      this is a test alert
    </div>
    <div
      className="__rta_close"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Alert should render error alert correctly 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(244, 67, 54, 0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      test
    </div>
  </div>
</div>
`;

exports[`Alert should render info alert correctly 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(33, 150, 243, 0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      test
    </div>
  </div>
</div>
`;

exports[`Alert should render success alert correctly 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(76, 175, 80, 0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      test
    </div>
  </div>
</div>
`;

exports[`Alert should render warning alert correctly 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(255, 152, 0, 0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      test
    </div>
  </div>
</div>
`;

exports[`Alert should render without close icon when closeButton prop equals false 1`] = `
<div
  className="__rta_container"
  style={
    Object {
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <div
    className="__rta_wrapper"
    style={
      Object {
        "backgroundColor": "rgba(76, 175, 80, 0.8)",
        "color": "white",
        "display": "flex",
        "fontFamily": "Lato,Helvetica Neue,Arial,Helvetica,sans-serif",
        "fontSize": "13px",
        "justifyContent": "flex-start",
        "minHeight": "40px",
        "overflow": "hidden",
        "width": "100%",
        "zIndex": 50,
      }
    }
  >
    <div
      className="__rta_icon"
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "padding": "10px",
          "width": "40px",
        }
      }
    >
      <img
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "margin": "0 auto",
          }
        }
      />
    </div>
    <div
      className="__rta_content"
      style={
        Object {
          "display": "flex",
          "padding": "10px",
          "width": "100%",
        }
      }
    >
      this is a test alert
    </div>
  </div>
</div>
`;
